tests/test_streaming.py

```diff
@@ -59,6 +59,65 @@ def test_dataloader_epoch_size_no_streams(local_remote_dir: Tuple[str,
             assert samples_seen == epoch_size
 
 
+@pytest.mark.parametrize('batch_size', [4])
+@pytest.mark.parametrize('seed', [2222])
+@pytest.mark.parametrize('shuffle', [True])
+@pytest.mark.parametrize('drop_last', [False, True])
+@pytest.mark.parametrize('num_workers', [3, 6])
+@pytest.mark.parametrize('num_canonical_nodes', [4, 8])
+@pytest.mark.usefixtures('local_remote_dir')
+def test_dataloader_per_stream_batching(local_remote_dir: Tuple[str, str], batch_size: int,
+                                        seed: int, shuffle: bool, drop_last: bool,
+                                        num_workers: int, num_canonical_nodes: int):
+    # create mock datasets for 2 streams. Second one has 1.5x the samples
+    local, remote = local_remote_dir
+    local1 = os.path.join(local, 'stream1')
+    local2 = os.path.join(local, 'stream2')
+    remote1 = os.path.join(remote, 'stream1')
+    remote2 = os.path.join(remote, 'stream2')
+
+    # stream 1 has samples 0->600
+    convert_to_mds(out_root=remote1,
+                   dataset_name='sequencedataset',
+                   num_samples=200,
+                   size_limit=1 << 8)
+    # stream 2 has samples 600 and above. This lets us differentiate between the samples from each stream
+    convert_to_mds(out_root=remote2,
+                   dataset_name='sequencedataset',
+                   num_samples=300,
+                   offset=600,
+                   size_limit=1 << 8)
+
+    stream1 = Stream(local=local1, remote=remote1)
```

Comment:
Does other streams args `proportion`, `choose`, and `repeat` have any impact or in other words, does the behavior still remain intact ? Does you think adding a test case with other stream args is valuable or no ?

Reply:
That logic is unchanged by the new batching methods, so i don't think it's worth to test. Before batching methods are applied, the `proportion`, `choose`, and `repeat` of each stream is already calculated, and the batching methods use the number of samples per stream corresponding to these param values.